### ISSUE TITLE
False Positive in LoopsShouldNotBeInfiniteCheck : support 'yield'

### ISSUE THREAD

**Issue by fmallet** [SonarSource/SonarJS#674]
[RSPEC-2189](https://jira.sonarsource.com/browse/RSPEC-2189)

`yield` should be supported in the CFG.

```
function * generator() {
    while(true) { // False positive here
        foo();
        yield;
    }
}
```

As soon as this is implemented we should put this in an Exceptions section on the rule's description.

**Comment by RWOverdijk** [comment]
👍

**Comment by usmonster** [comment]
Hello! Is this planned? It's a useful rule, except for this limitation, which means we exclude it from our quality profiles. Thanks!

**Comment by zanettin** [comment]
would love to get this fixed as well. forces use to not use generators/exclude them from the profiles. any plans do fix that?

**Comment by vilchik-elena** [comment]
Hey! This rule is still based on Java-based frontend we plan to drop. When migrating to ESLint-based frontend we will make sure this FP is not there anymore.
